forms-flow-web/src/apiManager/services/processServices.js

```diff
@@ -90,63 +90,45 @@ export const fetchAllBpmProcesses = (excludeInternal = true, ...rest) => {
   };
 };
 
-export const fetchAllBpmDeployments = (...rest) => {
-  const done = rest.length ? rest[0] : () => {};
-  return (dispatch) => {
+export const fetchAllBpmDeployments = () => {
+  return async (dispatch) => {
     // eslint-disable-next-line max-len
-    httpGETRequest(API.GET_BPM_DEPLOYMENTS, {}, UserService.getToken(), true)
-      .then(async (res) => {
-        if (res?.data) {
-          // GET Resources
-          const resourceIdList = res.data.map(async (deployment) => {
-            const apiUrlDeployementId = replaceUrl(
-              API.GET_BPM_DEPLOYMENT_RESOURCES,
-              "<deployment_id>",
-              deployment.id
-            );
-            // eslint-disable-next-line max-len
-            const resources = await httpGETRequest(
-              apiUrlDeployementId,
-              {},
-              UserService.getToken(),
-              true
-            );
-            deployment.isExecutable = false;
-            deployment.deploymentId = resources.data[0].deploymentId;
-            deployment.resourceId = resources.data[0].id;
-            deployment.deploymentName = resources.data[0].name;
-            deployment.key = resources.data[0].name.slice(0, -5); // remove .bpmn
-
-            return deployment;
-          });
-          const updatedResourceList = await Promise.all(resourceIdList);
+    httpGETRequest(
+      API.GET_BPM_DEPLOYMENTS + `?sortBy=deploymentTime&sortOrder=desc`,
+      {},
+      UserService.getToken(),
+      true
+    )
+      .then((res) => {
+        let deploymentList = [];
+        let numDeployments = 0;
 
-          // GET Diagrams
-          const diagramList = updatedResourceList.map(async (deployment) => {
-            let apiUrlDiagramList = replaceUrl(
-              API.GET_BPM_DEPLOYMENT_DIAGRAM,
-              "<deployment_id>",
-              deployment.deploymentId
-            );
-            apiUrlDiagramList = replaceUrl(
-              apiUrlDiagramList,
-              "<resource_id>",
-              deployment.resourceId
-            );
-            // eslint-disable-next-line max-len
-            const diagram = await httpGETRequest(
-              apiUrlDiagramList,
-              {},
-              UserService.getToken(),
-              true
-            );
-            deployment.diagram = diagram.data;
-            return deployment;
+        if (res?.data) {
+          // Get Deployments
+          res.data.map((deployment) => {
+            // Get Resources
+            getBpmDeploymentResources(deployment).then((resourceList) => {
+              resourceList.resources.map((resource) => {
```

Comment:
As mentioned on connect avoid fetching all the resources/diagram initially. Only when user clicks a deployment fetch it's corresponding details. Also @auslin-aot check for an optimum solution say if we can use the Id from deployment directly to get diagram , so to avoid multiple  api calls